src/python/pybind11/src/pyxrt.cpp

```diff
@@ -33,263 +33,262 @@ namespace py = pybind11;
-
-py::enum_<ert_cmd_state>(m, "ert_cmd_state")
-    .value("ERT_CMD_STATE_NEW", ert_cmd_state::ERT_CMD_STATE_NEW)
-    .value("ERT_CMD_STATE_QUEUED", ert_cmd_state::ERT_CMD_STATE_QUEUED)
-    .value("ERT_CMD_STATE_COMPLETED", ert_cmd_state::ERT_CMD_STATE_COMPLETED)
-    .value("ERT_CMD_STATE_ERROR", ert_cmd_state::ERT_CMD_STATE_ERROR)
-    .value("ERT_CMD_STATE_ABORT", ert_cmd_state::ERT_CMD_STATE_ABORT)
-    .value("ERT_CMD_STATE_SUBMITTED", ert_cmd_state::ERT_CMD_STATE_SUBMITTED)
-    .value("ERT_CMD_STATE_TIMEOUT", ert_cmd_state::ERT_CMD_STATE_TIMEOUT)
-    .value("ERT_CMD_STATE_NORESPONSE", ert_cmd_state::ERT_CMD_STATE_NORESPONSE)
-    .value("ERT_CMD_STATE_SKERROR", ert_cmd_state::ERT_CMD_STATE_SKERROR)
-    .value("ERT_CMD_STATE_SKCRASHED", ert_cmd_state::ERT_CMD_STATE_SKCRASHED)
-    .value("ERT_CMD_STATE_MAX", ert_cmd_state::ERT_CMD_STATE_MAX);
-
-py::enum_<xrt::info::device>(m, "xrt_info_device")
-    .value("bdf", xrt::info::device::bdf)
-    .value("interface_uuid", xrt::info::device::interface_uuid)
-    .value("kdma", xrt::info::device::kdma)
-    .value("max_clock_frequency_mhz", xrt::info::device::max_clock_frequency_mhz)
-    .value("m2m", xrt::info::device::m2m)
-    .value("name", xrt::info::device::name)
-    .value("nodma", xrt::info::device::nodma)
-    .value("offline", xrt::info::device::offline)
-    .value("electrical", xrt::info::device::electrical)
-    .value("thermal", xrt::info::device::thermal)
-    .value("mechanical", xrt::info::device::mechanical)
-    .value("memory", xrt::info::device::memory)
-    .value("platform", xrt::info::device::platform)
-    .value("pcie_info", xrt::info::device::pcie_info)
-    .value("host", xrt::info::device::host)
-    .value("dynamic_regions", xrt::info::device::dynamic_regions);
+        .value("XCL_BO_SYNC_BO_FROM_DEVICE", xclBOSyncDirection::XCL_BO_SYNC_BO_FROM_DEVICE)
+        .value("XCL_BO_SYNC_BO_GMIO_TO_AIE", xclBOSyncDirection::XCL_BO_SYNC_BO_GMIO_TO_AIE)
+        .value("XCL_BO_SYNC_BO_AIE_TO_GMIO", xclBOSyncDirection::XCL_BO_SYNC_BO_AIE_TO_GMIO);
+
+    py::enum_<ert_cmd_state>(m, "ert_cmd_state", "Kernel execution status")
+        .value("ERT_CMD_STATE_NEW", ert_cmd_state::ERT_CMD_STATE_NEW)
+        .value("ERT_CMD_STATE_QUEUED", ert_cmd_state::ERT_CMD_STATE_QUEUED)
+        .value("ERT_CMD_STATE_COMPLETED", ert_cmd_state::ERT_CMD_STATE_COMPLETED)
+        .value("ERT_CMD_STATE_ERROR", ert_cmd_state::ERT_CMD_STATE_ERROR)
+        .value("ERT_CMD_STATE_ABORT", ert_cmd_state::ERT_CMD_STATE_ABORT)
+        .value("ERT_CMD_STATE_SUBMITTED", ert_cmd_state::ERT_CMD_STATE_SUBMITTED)
+        .value("ERT_CMD_STATE_TIMEOUT", ert_cmd_state::ERT_CMD_STATE_TIMEOUT)
+        .value("ERT_CMD_STATE_NORESPONSE", ert_cmd_state::ERT_CMD_STATE_NORESPONSE)
+        .value("ERT_CMD_STATE_SKERROR", ert_cmd_state::ERT_CMD_STATE_SKERROR)
+        .value("ERT_CMD_STATE_SKCRASHED", ert_cmd_state::ERT_CMD_STATE_SKCRASHED)
+        .value("ERT_CMD_STATE_MAX", ert_cmd_state::ERT_CMD_STATE_MAX);
```

Comment:
I don't think we should expose ERT_CMD_STATE_MAX, it should have no use in host code.  In general, I don't really think we should have an enumerator exposed that can change value from release to release; maybe not so bad python but in a compiled application relying on ERT_CMD_STATE_MAX would not be good.  We should probably get rid of this enumerator.

Reply:
We will need the enums as that is the status of a few kernel methods. I will create another PR to remove the "ERT_CMD_STATE_MAX".

Reply:
@sonal.  I was not suggesting to remove all the enumerators, only the ERT_CMD_STATE_MAX which shouldn't be there in the first place in ert.h.